packages/models-library/src/models_library/string_types.py

```diff
@@ -0,0 +1,136 @@
+import re
+from typing import Annotated, Final, TypeAlias
+
+import annotated_types
+from pydantic import (
+    AfterValidator,
+    StringConstraints,
+)
+from pydantic_core import PydanticCustomError
+
+from .utils.common_validators import trim_string_before
+
+# --- shared heuristics ---
+MIN_DESCRIPTION_LENGTH: Final[int] = 3
+MAX_DESCRIPTION_LENGTH: Final[int] = 5000
+MAX_NAME_LENGTH: Final[int] = 100
+
+_SHORT_TRUNCATED_STR_MAX_LENGTH: Final[int] = 600
+_LONG_TRUNCATED_STR_MAX_LENGTH: Final[int] = 65536  # same as github descriptions
+
+_SQL_INJECTION_PATTERN: Final[re.Pattern] = re.compile(
+    r"(?i)(?:\b(?:SELECT|INSERT|UPDATE|DELETE|DROP|UNION|ALTER|CREATE|EXEC|TRUNCATE|MERGE|GRANT|REVOKE|COMMIT|ROLLBACK|DECLARE|CAST|CONVERT)\b|--|;|/\*|\*/|')",
+)
+_JS_INJECTION_PATTERN: Final[re.Pattern] = re.compile(
+    r"(?i)<(?:script|iframe|object|embed)(?:\s[^>]{0,100})?>|</(?:script|iframe|object|embed)>|<link(?:\s[^>]{0,200})?href\s*=\s*[\"']?\s*javascript:|(?:vb|java)script:|data:text/html|&#(?:x6A|106);avascript:|<(?:img|svg)(?:\s[^>]{0,200})?on\w+\s*=|on[a-z]+\s*=",
+)
+STRING_UNSAFE_CONTENT_ERROR_CODE: Final[str] = "string_unsafe_content"
```

Comment:
question: did you check these mega regex against "catastrophic backtracking"? Cause this can be a worse remerdy than the initial problem.
see https://www.sonarsource.com/blog/vulnerable-regular-expressions-javascript/ for example